cmd/edit/cluster/cmd.go

```diff
@@ -500,6 +518,47 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	/*******  AdditionalAllowedPrincipals *******/
+	updateAdditionalAllowedPrincipals := false
+	if additionalAllowedPrincipals != nil {
```

Comment:
Do we need to consider empty string[] scenario?

Reply:
Not here because empty string[] is a valid input. It means we will need to update the additional allowed principals by setting it to empty.